### PR TITLE
TS-3917: Sending only SETTINGS_INITIAL_WINDOW_SIZE in SETTINGS Frame

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

**Comment by zwoop** [comment]
+1

File: proxy/http2/HTTP2.cc [hunk 1]
```diff
@@ -727,7 +727,7 @@ http2_decode_header_blocks(HTTPHdr *hdr, const uint8_t *buf_start, const uint8_t
 }
 
 // Initialize this subsystem with librecords configs (for now)
-uint32_t Http2::max_concurrent_streams = 100;
+uint32_t Http2::max_concurrent_streams = 4294967295;
```
Comment (masaori335) [inline proxy/http2/HTTP2.cc]: This variable is override by `proxy.config.http2.max_concurrent_streams_in` in [Http2::init()](https://github.com/apache/trafficserver/blob/master/proxy/http2/HTTP2.cc#L743).
IMO, where we should change is `HTTP2_MAX_CONCURRENT_STREAMS` in [HTTP2.h#L57](https://github.com/apache/trafficserver/blob/master/proxy/http2/HTTP2.h#L57)

**Comment by bryancall** [comment]
@masaori335 I am pretty sure the compare when setting the settings uses this value to compare with.

**Comment by masaori335** [comment]
@bryancall Oh really? I'll check.

**Comment by bryancall** [comment]
@masaori335 You are right :)

**Comment by masaori335** [comment]
@bryancall OK :)

**Pull request closed without merging**